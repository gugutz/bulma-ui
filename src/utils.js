export function classnames(classes, rules) {
  return Object.keys(rules)
    .reduce((acc, rule) => acc + (rules[rule] ? ' ' + rule : ''), classes)
    .replace(/^\s+|\s+$/g, '')
}

export function range(start, end) {
  const acc = []

  for (let i = start; end - i > 0; i++) {
    acc.push(i)
  }

  return acc
}
