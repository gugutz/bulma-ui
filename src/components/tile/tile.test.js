import React from 'react'
import renderer from 'react-test-renderer'
import Box from '../box/box'
import {range} from '../../utils'
import Tile from './tile'

describe('<Tile /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Tile isAncestor>
          <Tile isVertical size={8}>
            <Tile>
              <Tile isParent isVertical>
                <Tile isChild>
                  <Box>Put any content you want</Box>
                </Tile>
                <Tile isChild>
                  <Box>Put any content you want</Box>
                </Tile>
              </Tile>
              <Tile isParent>
                <Tile isChild>
                  <Box>Put any content you want</Box>
                </Tile>
              </Tile>
            </Tile>
            <Tile isParent>
              <Tile isChild>
                <Box>Put any content you want</Box>
              </Tile>
            </Tile>
          </Tile>
          <Tile isParent>
            <Tile isChild>
              <Box>Put any content you want</Box>
            </Tile>
          </Tile>
        </Tile>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const sizes = range(1, 13)

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer
        .create(
          <Tile size={size}>
            <Box>Put any content you want</Box>
          </Tile>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
