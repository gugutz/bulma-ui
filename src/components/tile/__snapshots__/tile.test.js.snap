// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> component should render correctly 1`] = `
<div
  className="tile is-ancestor"
>
  <div
    className="tile is-vertical is-8"
  >
    <div
      className="tile"
    >
      <div
        className="tile is-parent is-vertical"
      >
        <div
          className="tile is-child"
        >
          <div
            className="box"
          >
            Put any content you want
          </div>
        </div>
        <div
          className="tile is-child"
        >
          <div
            className="box"
          >
            Put any content you want
          </div>
        </div>
      </div>
      <div
        className="tile is-parent"
      >
        <div
          className="tile is-child"
        >
          <div
            className="box"
          >
            Put any content you want
          </div>
        </div>
      </div>
    </div>
    <div
      className="tile is-parent"
    >
      <div
        className="tile is-child"
      >
        <div
          className="box"
        >
          Put any content you want
        </div>
      </div>
    </div>
  </div>
  <div
    className="tile is-parent"
  >
    <div
      className="tile is-child"
    >
      <div
        className="box"
      >
        Put any content you want
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 1 1`] = `
<div
  className="tile is-1"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 2 1`] = `
<div
  className="tile is-2"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 3 1`] = `
<div
  className="tile is-3"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 4 1`] = `
<div
  className="tile is-4"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 5 1`] = `
<div
  className="tile is-5"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 6 1`] = `
<div
  className="tile is-6"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 7 1`] = `
<div
  className="tile is-7"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 8 1`] = `
<div
  className="tile is-8"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 9 1`] = `
<div
  className="tile is-9"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 10 1`] = `
<div
  className="tile is-10"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 11 1`] = `
<div
  className="tile is-11"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;

exports[`<Tile /> component should support size prop with 12 1`] = `
<div
  className="tile is-12"
>
  <div
    className="box"
  >
    Put any content you want
  </div>
</div>
`;
